test/Microsoft.TestPlatform.AcceptanceTests/PostProcessingTests.cs

```diff
@@ -0,0 +1,122 @@
+
+        // Set datacollector parameters
+        XElement runSettingsXml = XElement.Load(runSettings);
+        runSettingsXml.Element("DataCollectionRunSettings")
+            .Element("DataCollectors")
+            .Element("DataCollector")
+            .Add(new XElement("Configuration", new XElement("MergeFile", "MergedFile.txt")));
+        runSettingsXml.Save(runSettings);
+
+        // Build and run tests like msbuild
+        Parallel.For(0, 5, i =>
+        {
+            string projectFolder = Path.Combine(tempDir.Path, i.ToString());
+            ExecuteApplication(GetConsoleRunnerPath(), $"new mstest -o {projectFolder}", out string stdOut, out string stdError, out int exitCode);
+            Assert.AreEqual(exitCode, 0);
+            ExecuteApplication(GetConsoleRunnerPath(), $"build {projectFolder} -c release", out stdOut, out stdError, out exitCode);
+            Assert.AreEqual(exitCode, 0);
+
+            string testContainer = Directory.GetFiles(Path.Combine(projectFolder, "bin"), $"{i}.dll", SearchOption.AllDirectories).Single();
+
+            ExecuteVsTestConsole($"{testContainer} --Collect:\"SampleDataCollector\" --TestAdapterPath:\"{extensionsPath}\" --ResultsDirectory:\"{Path.GetDirectoryName(testContainer)}\" --Settings:\"{runSettings}\" --ArtifactsProcessingMode-Collect --TestSessionCorrelationId:\"{correlationSessionId}\" --Diag:\"{tempDir.Path + '/'}\"", out stdOut, out stdError, out exitCode,
+                new Dictionary<string, string>() { { "VSTEST_FEATURE_ARTIFACTS_POSTPROCESSING", "1" } });
+            Assert.AreEqual(exitCode, 0);
+        });
+
+        // Post process artifacts
+        ExecuteVsTestConsole($"--ArtifactsProcessingMode-PostProcess --TestSessionCorrelationId:\"{correlationSessionId}\" --Diag:\"{tempDir.Path + '/'}\"", out string stdOut, out string stdError, out int exitCode,
+                new Dictionary<string, string>() { { "VSTEST_FEATURE_ARTIFACTS_POSTPROCESSING", "1" } });
+        Assert.AreEqual(exitCode, 0);
+
+        using StringReader reader = new(stdOut);
+        Assert.AreEqual(string.Empty, reader.ReadLine().Trim());
+        Assert.AreEqual("Attachments:", reader.ReadLine().Trim());
+        string mergedFile = reader.ReadLine().Trim();
+        Assert.AreEqual(string.Empty, reader.ReadLine().Trim());
+        Assert.IsNull(reader.ReadLine());
+
+        var fileContent = new List<string>();
+        using var streamReader = new StreamReader(mergedFile);
+        while (!streamReader.EndOfStream)
+        {
+            string line = streamReader.ReadLine();
+            Assert.IsTrue(line.StartsWith("SessionEnded_Handler_"));
+            fileContent.Add(line);
+        }
+
+        Assert.AreEqual(5, fileContent.Distinct().Count());
+    }
+
+    private static string GetRunsettingsFilePath(string resultsDir)
+    {
+        var runsettingsPath = Path.Combine(resultsDir, "test_" + Guid.NewGuid() + ".runsettings");
```

Comment:
I don't think the guid is required as we have a unique folder name already.

Reply:
Copy pasted from another tests with runsettings. 
That guid is useful if we'll update test with different runsettings for different projects. I'll keep as-is.